godtools/ViewControllers/Tract/TractViewControllerContent.swift

```diff
@@ -47,39 +47,30 @@ extension TractViewController {
     
     func reloadPagesViews() {
         let range = getRangeOfViews()
-        let firstTag = (self.pagesViews.first?.tag)! - self.viewTagOrigin
-        let lastTag = (self.pagesViews.last?.tag)! - self.viewTagOrigin
         let width = self.containerView.frame.size.width
         let height = self.containerView.frame.size.height
+        let lastPosition = self.totalPages() - 1
+        var tmpPagesViews = [BaseTractView?](repeating: nil, count: 10)
         
-        if firstTag < range.start {
-            for _ in 0...firstTag {
-                self.pagesViews.first?.removeFromSuperview()
-                self.pagesViews.removeFirst()
-            }
-            
-        } else if firstTag > range.start {
-            var position = 0
-            for _ in range.start...firstTag {
-                let view = buildPage(range.start + position, width: width, height: height)
-                self.pagesViews.insert(view, at: position)
+        for position in range.start...range.end {
+            let pageView = self.pagesViews[position]
```

Comment:
This could be shortened to use `if let` which I think unwraps for you
````
if let pageView = self.pageViews[position] {
 tmpPageViews[position] = pageView 
} else {
 //el mismo code aqui
}